### PR TITLE
Update docs for 2.1.0

### PR DESCRIPTION
**Pull request by ghostdogpr**
To simplify things a little further, I moved the `graphQL` function to the `caliban` package. The old one is still there but deprecated.

### CODE REVIEW

File: vuepress/docs/docs/schema.md [hunk 1]
```diff
@@ -164,6 +139,55 @@ An interface will be created with all the fields that are common to the case cla
...
+
+implicit val schemaForMyClass: Schema[Any, MyClass] = Schema.gen
+```
+  </code-block>
+  <code-block title="Scala 3">
+
+```scala
+import caliban.schema.Schema
+
+case class MyClass(field: String) derives Schema.SemiAuto
```
Comment (kyri-petrou) [inline vuepress/docs/docs/schema.md]: Should we also mention `derives Schema.Auto`? I found it particularly useful for enums where it's cumbersome to create a custom schema for each individual enum value

File: vuepress/docs/about/README.md
Comment (kyri-petrou) [inline vuepress/docs/about/README.md:1]: Should we mention somewhere here that Caliban 2.1.0+ requires JDK 11?

**Review by kyri-petrou — COMMENTED** [review]
LGTM - just some minor comments regarding docs

**Comment by frekw** [comment]
> I moved the graphQL function to the caliban package

This is actually a huge quality of life change. I always forget where it lives and Metals can never seem to figure it out :D

File: vuepress/docs/docs/README.md [hunk 2]
```diff
@@ -12,32 +12,32 @@ The design principles of Caliban are the following:
...
-libraryDependencies += "com.github.ghostdogpr" %% "caliban-federation" % "2.0.2" // interop with apollo federation
+```scala
+"com.github.ghostdogpr" %% "caliban-http4s"     % "2.1.0" // routes for http4s
+"com.github.ghostdogpr" %% "caliban-akka-http"  % "2.1.0" // routes for akka-http
+"com.github.ghostdogpr" %% "caliban-play"       % "2.1.0" // routes for play
+"com.github.ghostdogpr" %% "caliban-zio-http"   % "2.1.0" // routes for zio-http
+"com.github.ghostdogpr" %% "caliban-cats"       % "2.1.0" // interop with cats effect
+"com.github.ghostdogpr" %% "caliban-monix"      % "2.1.0" // interop with monix
+"com.github.ghostdogpr" %% "caliban-tapir"      % "2.1.0" // interop with tapir
+"com.github.ghostdogpr" %% "caliban-federation" % "2.1.0" // interop with apollo federation
```
Comment (frekw) [inline vuepress/docs/docs/README.md:31]:
```suggestion
"com.github.ghostdogpr" %% "caliban-federation" % "2.1.0" // interop with apollo federation
"com.github.ghostdogpr" %% "caliban-tracing" % "2.1.0" // interop with zio-telemetry
```

File: vuepress/docs/docs/schema.md [hunk 3]
```diff
@@ -164,6 +139,55 @@ An interface will be created with all the fields that are common to the case cla
 
 If you prefer to have a `Union` type instead of an `Enum`, even when the sealed trait contains only objects, add the `@GQLUnion` annotation.
 
+## Case classes and sealed traits
+
+The transformation between Scala types and GraphQL types is handled by a typeclass named `Schema`. As mentioned earlier, caliban provides instances of `Schema` for all basic Scala types, but inevitably you will need to support your own types, in particular case classes and sealed traits.
```
Comment (frekw) [inline vuepress/docs/docs/schema.md]:
```suggestion
The transformation between Scala types and GraphQL types is handled by a typeclass named `Schema`. As mentioned earlier, Caliban provides instances of `Schema` for all basic Scala types, but inevitably you will need to support your own types, in particular case classes and sealed traits.
```

File: vuepress/docs/docs/schema.md [hunk 4]
```diff
@@ -164,6 +139,55 @@ An interface will be created with all the fields that are common to the case cla
...
+
+Caliban is able to generate instances of `Schema` for case classes and sealed traits. You have 2 choices for doing that: auto derivation and semi-auto derivation.
```
Comment (frekw) [inline vuepress/docs/docs/schema.md]:
```suggestion
Caliban is able to generate instances of `Schema` for case classes and sealed traits. You have two choices for doing that: auto derivation and semi-auto derivation.
```

File: vuepress/docs/docs/schema.md [hunk 5]
```diff
@@ -164,6 +139,55 @@ An interface will be created with all the fields that are common to the case cla
...
+
+### Auto derivation
+Auto derivation is achieved easily by adding the following import:
+```scala mdoc:silent
+import caliban.schema.Schema.auto._
+```
+Using this import, caliban will generate a `Schema` instance for all the case classes and sealed traits that are found inside your resolver.
```
Comment (frekw) [inline vuepress/docs/docs/schema.md]:
```suggestion
Using this import, Caliban will generate a `Schema` instances for all the case classes and sealed traits inside your resolver.
```

**Comment by frekw** [comment]
Great job, I think this makes the derivation docs a lot more helpful since we both talk more about what `Schema` is as well as showcase the different usage patterns :)

File: vuepress/docs/about/README.md
Comment (ghostdogpr) [reply vuepress/docs/about/README.md:1]: I was thinking to mention it in the release notes 🤔

**Comment by ghostdogpr** [comment]
Applied review suggestions

**Review by frekw — APPROVED** [review]

File: vuepress/docs/about/README.md
Comment (kyri-petrou) [reply vuepress/docs/about/README.md:1]: Sounds good. I was thinking perhaps for the benefit of new users but then again it's highly unlikely someone starting a new project they'll use JDK8.

**Review by kyri-petrou — APPROVED** [review]

**Pull request merged**